hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -316,9 +316,9 @@ public boolean bind(final ClientEndpoint endpoint) {
             // On such a case, `ClientEngine#connectionRemoved` will not be called for this connection since
             // we did not register the connection.
             // Endpoint removal logic(inside `ClientEngine#connectionRemoved`) will not be able to run, instead endpoint
-            // will be cleaned up by ClientHearbeatMonitor#cleanupEndpointsWithDeadConnections later.
-            if (conn.getRemoteAddress() != null) {
-                node.getServer().getConnectionManager(CLIENT).register(conn.getRemoteAddress(), conn);
+            // will be cleaned up by ClientHeartbeatMonitor#cleanupEndpointsWithDeadConnections later.
+            if (conn.getRemoteAddress() != null && endpoint.getUuid() != null) {
```

Comment:
We have this value after the client endpoint is authenticated, it has no chance of being null in the production code. But if the some staff is mocked, it can be.